lib/activerecord-import/adapters/postgresql_adapter.rb

```diff
@@ -30,6 +31,67 @@ def post_sql_statements( table_name, options ) # :nodoc:
     end
   end
 
+  # Add a column to be updated on duplicate key update
+  def add_column_for_on_duplicate_key_update( column, options={} ) # :nodoc:
+    if options[:on_duplicate_key_update].is_a?( Hash )
+      columns = options[:on_duplicate_key_update].fetch(:columns, [])
+      case columns
+      when Array then columns << column.to_sym unless columns.include?( column.to_sym )
+      when Hash then columns[column.to_sym] = column.to_sym
+      end
+    end
+  end
+
+  # Returns a generated ON CONFLICT DO UPDATE statement given the passed
+  # in +options+.
+  def sql_for_on_duplicate_key_update( table_name, options={} ) # :nodoc:
+    return unless options.is_a?( Hash )
+
+    key = Array( options[:key] ).join( ', ' )
+    raise ArgumentError, 'Attribute :key not found' if key.empty?
```

Comment:
It looks like `:key` represents the [ON CONFLICT](http://www.postgresql.org/docs/9.5/static/sql-insert.html#SQL-ON-CONFLICT)  `conflict_target`. If that's the case can we make this optional?